### PR TITLE
Replace CaseType's own XML encoding function

### PR DESCRIPTION
**Pull request by artfulrobot**
The problem is detailed at https://lab.civicrm.org/dev/core/-/issues/1719

In a nutshell:

1. Creating a case type crashes if the `name` of a case status is not valid XML content. e.g. if a name has `&` in it, but other special characters would also cause problems.

2. It dies because it fails to properly encode the value.

3. It uses its own encoding functions instead of a proper XML encoder.

This PR:

1. replaces the homespun XML encoder with php's own built in one, and properly encodes everything.

2. adds a test fixture for the case that `&` is in a status name.

As a side effect, I believe it fixes 1719

### CODE REVIEW

**Comment by mattwire** [comment]
ping @demeritcowboy

**Comment by demeritcowboy** [comment]
The test failure is real but seems like it's mostly just the utf->UTF thing. There's also a whitespace difference between what came out before and after, but that seems mostly clerical. If anybody is relying on specific whitespace in their xml they shouldn't be, but the test is complaining.

File: CRM/Case/BAO/CaseType.php [hunk 1]
```diff
@@ -96,119 +96,156 @@ protected function assignTestValue($fieldName, &$fieldDef, $counter) {
...
-    $xmlFile .= '</CaseType>';
-
-    return $xmlFile;
-  }
+    // CaseType
+    $xw->fullEndElement();
+    $xw->endDocument();
 
-  /**
-   * Ugh. This shouldn't exist. Use a real XML-encoder.
```
Comment (colemanw) [inline CRM/Case/BAO/CaseType.php:195]: And now it doesn't! Hooray :tada:

**Comment by colemanw** [comment]
Code is an excellent improvement, and test coverage is solid.

**Pull request merged**